include/seqan3/utility/tuple/common_tuple.hpp

```diff
@@ -6,37 +6,1030 @@
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_assignable_v<Types const, UTypes> && ...)
+    constexpr common_tuple const & operator=(std::pair<UTypes...> && other) const
+    {
+        [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            ((std::get<N>(*this) = std::get<N>(other)), ...);
+        } (std::index_sequence_for<Types...>{});
+
+        return *this;
+    }
+
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_assignable_v<Types const, UTypes const> && ...)
+    constexpr common_tuple const & operator=(std::pair<UTypes...> const && other) const
+    {
+        [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            ((std::get<N>(*this) = std::get<N>(other)), ...);
+        } (std::index_sequence_for<Types...>{});
+
+        return *this;
+    }
+    //!@}
+    //!\}
+
+    /*!\name Conversion to std::tuple.
+     * \{
+     */
+    //!@{ Converts to std::tuple
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_constructible_v<UTypes, Types &> && ...)
+    operator std::tuple<UTypes...>() &
+    {
+        return std::make_from_tuple<std::tuple<UTypes...>>(*this);
+    }
+
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_constructible_v<UTypes, Types const> && ...)
+    operator std::tuple<UTypes...>() const &
+    {
+        return std::make_from_tuple<std::tuple<UTypes...>>(*this);
+    }
+
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_constructible_v<UTypes, Types> && ...)
+    operator std::tuple<UTypes...>() &&
+    {
+        return std::make_from_tuple<std::tuple<UTypes...>>(std::move(*this));
+    }
+
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) && (std::is_constructible_v<UTypes, Types const> && ...)
+    operator std::tuple<UTypes...>() const &&
+    {
+        return std::make_from_tuple<std::tuple<UTypes...>>(std::move(*this));
+    }
+    //!@}
+    //!\}
+
+    /*!\name Comparison operators (common_tuple)
+     * \{
+     */
+    /*!\brief Checks whether `lhs` and `rhs` are equal.
+     * \tparam UTypes The types of the elements of `rhs`. Automatically deduced.
+     * \param lhs A common_tuple.
+     * \param rhs A common_tuple with possibly different element types.
+     * \returns A bool indicating the result of the comparison.
+     */
+    template <class... UTypes>
+        requires (sizeof...(Types) == sizeof...(UTypes)) &&
+            requires { requires (std::equality_comparable_with<Types, UTypes> && ...); } // Defer instantiation
+    constexpr friend bool operator==(common_tuple const & lhs, common_tuple<UTypes...> const & rhs)
+    {
+        static_assert((std::equality_comparable_with<Types, UTypes> && ...));
+        return lhs.as_base() == rhs.as_base();
+    }
```

Comment:
Note: Because this is a friend, I need to make sure the concept check is not a hard error if it cannot compile.

Apparently, the checks are short-circuited when it is a free function, but not when it is a friend (I assume that is a bug).

Hence, it will try instantiating something like 
```cpp
operator==(common_tuple<int, int> const & lhs, common_tuple<int> const & rhs)
```

When unfolding
```cpp
requires (std::equality_comparable_with<Types, UTypes> && ...)
```
the packs are not of the same length. This results in a hard compiler error.
Putting it into a `requires` expressions means that it will just return `false` instead of erroring.